mocked-api/handlers.ts

```diff
@@ -435,6 +435,120 @@ export const handlers = [
       )
     );
   }),
+  // update a processor
+  rest.put(
+    `${apiUrl}/bridges/:bridgeId/processors/:processorId`,
+    (req, res, ctx) => {
+      const { bridgeId, processorId } = req.params;
+      const { name, filters, transformationTemplate, source, action } =
+        req.body as ProcessorRequest;
+
+      const existingBridge = db.bridge.findFirst({
+        where: {
+          id: {
+            equals: bridgeId as string,
+          },
+        },
+      });
+
+      if (!existingBridge) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Bridge with id '${
+              bridgeId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      const processor = db.processor.findFirst({
+        where: {
+          id: {
+            equals: processorId as string,
+          },
+        },
+      });
+
+      if (!processor) {
+        return res(
+          ctx.status(404),
+          ctx.delay(apiDelay),
+          ctx.json({
+            ...error_not_found,
+            reason: `Processor with id '${
+              processorId as string
+            }' for customer 'XXXXXXXX' does not exist`,
+          })
+        );
+      }
+
+      const existingProcessors = db.processor.findMany({
+        where: {
+          name: {
+            equals: name,
+          },
+          bridge: {
+            id: {
+              equals: bridgeId as string,
+            },
+          },
+        },
+      });
+
+      if (existingProcessors.length > 1) {
```

Comment:
Oh, it actually means that you renamed the edited processor like another one.
Does it make sense to you?

Reply:
@manstis fixed with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/54/commits/75e9bc5a72d0cc2c1985803718907f95e5d38b48
I just noticed something that could be interesting also for @kelvah 
When updating a processor, its status will not be "ready" anymore, so the status loaded on the detail page will not reflect the flow on the "backend".
Since we don't have a refresh button here, do you suggest considering a polling mechanism (as a future improvement)?

Reply:
@vpellegrino not sure about what's the approach we should follow here. I guess polling? Let's discuss it separately